facet-json-write/src/serialize.rs

```diff
@@ -182,6 +198,13 @@ pub fn to_json<W: Write>(peek: Peek<'_>, writer: &mut W, indent: bool) -> io::Re
                             });
                         }
                     }
+                    Peek::Option(popt) => {
+                        if let Some(p) = popt.value() {
+                            peek_value_to_json(p.as_value(), writer)?;
+                        } else {
+                            write!(writer, "null")?;
+                        }
+                    }
```

Comment:
I also snuck in turning optionals into `null` in json-write, as i think it makes sense (but happy to remove it). I also kinda wish I would have called `peek_value_to_json((), writer)?;` to not have `write!(writer, "null")?;` in two places, but it wasn't immediately obvious how to construct a `PeekValue` for `()`, and I got lazy.